### DISCUSSION TITLE
gtk.css windows location

### DISCUSSION THREAD

**Discussion by mtbf99** [zim-desktop-wiki/zim-desktop-wiki#1714] [Q&A] [answered]
Hi all,

In the windows environment where should I put gtk.css file? (Windows Installer)

I have a a working gtk.css that was tested in linux but it's not working in widows.

I tested in the following dirs:

- AppData\Roaming\zim
- AppData\Roaming\zim\config\gtk-30.0
- C:\Program Files\Zim Desktop Wiki\etc\gtk-3.0
- C:\Program Files\Zim Desktop Wiki\share\themes\Default\gtk-3.0
- C:\Program Files\Zim Desktop Wiki\share\themes\Emacs\gtk-3.0

CSS code:

#zim-pageview text {
    background-color: #1A2E38;  /* changes the background color of the Zim's page editor */
    color: #BABABA; /* foreground text color */
  }

#zim-pageindex  {
    background-color: #1A2E38;  /* changes the background color of the Zim's page editor */
    color: #BABABA; /* foreground text color */
    font-size: 15px;
  }

#zim-toc-widget {
  color: #2b2b2b;  /* sets the color of the border line of the Table of Content plugin */
  opacity: 0.9;    /* give the ToC some nice transparency */
  font-size: 12px; /* ToC font size */
  }

#zim-toc-widget scrolledwindow treeview {
  color: #BABABA;  /* sets the color of the text within the Table of Content plugin */
  opacity: 1;
  }

Regards

**Comment by jaap-karssenberg** [comment]
What version of Zim do you have and what install did you use, regular or
portable?

Op zo 31 okt. 2021 13:28 schreef Carlos Silva ***@***.***>:

**Reply by mtbf99** [reply]
Version 0.74.2 and it's a regular installation (with the installer)

**Reply by mtbf99** [reply]
Finally i found the waldo and a new road block:
What i did:

1. Created the directory <zim_install_dir>\etc\gtk-3.0
2. Created the settings.ini file under the new directory with the following text:
`[Settings]`
`gtk-theme-name=<theme_name>`
3. Created the directory for the theme: <zim_install_dir>\share\themes\\<theme_name>\gtk-3.0 
4. Created the gtk.css file on the new directory.

Now pageindex as the right font size and colors but the rest is transparent 

![image](https://user-images.githubusercontent.com/1427771/139594430-3c1f7ae5-c3bb-40fe-986a-6f9f4c862352.png)
![image](https://user-images.githubusercontent.com/1427771/139594456-91237bc1-1517-47ed-9935-570ee7daac06.png)

I'm doing this to increase the pageindex font size  because i have a laptop with high resolution in a 13" size screen.

Would be nice to have an option to change the page index font or apply the custom font option to the page index also.

Thank you Jaap for your great tool.

**Comment by jaap-karssenberg** [comment, marked as answer]
Hi Carlos,

Good you found it already. For future reference, this page can help:
https://zim-wiki.org/manual/Help/Config_Files.html

I think the transparency is due to something in your CSS. Probably need to
check where you change transparency and whether it doesn't apply to broadly.

Regards,

Jaap

**Reply by rowsof** [reply]
Hey there! Just wanted to note that the transparency issue above occurs when referencing a custom gtk.css file, even if it's completely empty. I think basically all css formatting is lost, as some other issues are apparent such as lack of padding around navbar buttons. My guess is the custom gtk.css file is overwriting the default gtk.css settings completely, rather than cascading. Unfortunately I have no idea where to find those default settings to copy them over as a starting point.

Would also suggest moving mtbf99's steps to the windows section of https://zim-wiki.org/manual/Help/Config_Files.html, since its instructions for modifying css on windows appear to be incorrect.

I am using a portable install, on 0.75.2 windows build

**Reply by dehesselle** [reply]
> Unfortunately I have no idea where to find those default settings to copy them over as a starting point.

You need to look at the sources for that: https://gitlab.gnome.org/GNOME/gtk/-/tree/gtk-3-24/gtk/theme/Adwaita

**Reply by rowsof** [reply]
Oooh, this is awesome. TY!
